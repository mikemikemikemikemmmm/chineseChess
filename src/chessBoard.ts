import Konva from "konva"
import { startX, startY, gridWidth, chessBoardHeightGridNum, chessBoardWidthGridNum } from "./const"
const strokeWidth = 4
const strokeColor = 'black'
const chessBoardLayer = new Konva.Layer({
    listening: false,
    x: startX,
    y: startY,
})
for (let y = 0; y < chessBoardHeightGridNum; y++) {
    for (let x = 0; x < chessBoardWidthGridNum; x++) {
        chessBoardLayer.add(new Konva.Rect({
            width: gridWidth, height: gridWidth,
            stroke: strokeColor,
            strokeWidth,
            x: gridWidth * x,
            y: gridWidth * y
        }))
    }
}

chessBoardLayer.add(new Konva.Rect({
    width: gridWidth * chessBoardWidthGridNum - strokeWidth,
    height: gridWidth - strokeWidth,
    fill: 'white',
    x: strokeWidth / 2,
    y: gridWidth * 4 + strokeWidth / 2
}))
chessBoardLayer.add(new Konva.Text({
    width: gridWidth * chessBoardWidthGridNum - strokeWidth,
    height: gridWidth - strokeWidth,
    text: '楚河   漢界',
    fontSize: gridWidth * 2 / 3,
    align: 'center',
    verticalAlign: 'middle',
    fill: strokeColor,
    x: strokeWidth / 2,
    y: gridWidth * 4 + strokeWidth / 2,
    strokeWidth: 2,
    stroke: 'black'
}))
chessBoardLayer.add(new Konva.Line({
    stroke: strokeColor,
    points: [
        gridWidth * 3,
        0,
        gridWidth * 5,
        gridWidth * 2,
    ],
    strokeWidth
}))
chessBoardLayer.add(new Konva.Line({
    stroke: strokeColor,
    points: [
        gridWidth * 3,
        0,
        gridWidth * 5,
        gridWidth * 2,
    ],
    strokeWidth
}))
chessBoardLayer.add(new Konva.Line({
    stroke: strokeColor,
    points: [
        gridWidth * 5,
        0,
        gridWidth * 3,
        gridWidth * 2,
    ],
    strokeWidth
}))
chessBoardLayer.add(new Konva.Line({
    stroke: strokeColor,
    points: [
        gridWidth * 5,
        gridWidth * 7,
        gridWidth * 3,
        gridWidth * 9,
    ],
    strokeWidth
}))
chessBoardLayer.add(new Konva.Line({
    stroke: strokeColor,
    points: [
        gridWidth * 3,
        gridWidth * 7,
        gridWidth * 5,
        gridWidth * 9,
    ],
    strokeWidth
}))
const factionText = new Konva.Text({
    width: gridWidth * chessBoardWidthGridNum,
    height: gridWidth,
    fontSize: gridWidth * 2 / 3,
    align: 'center',
    verticalAlign: 'middle',
    fill: strokeColor,
    y: -gridWidth * 1.5,
    strokeWidth: 2,
})
chessBoardLayer.add(factionText)
export {factionText,chessBoardLayer}