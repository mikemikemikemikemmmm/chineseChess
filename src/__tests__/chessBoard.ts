import { chessBoardLayer } from "../chessBoard"
describe("test chessBoard snapshot", () => {
    test('with canva', () => {
        const dataUrl = chessBoardLayer.toDataURL()
        expect(dataUrl).toMatchSnapshot()
    })
})
