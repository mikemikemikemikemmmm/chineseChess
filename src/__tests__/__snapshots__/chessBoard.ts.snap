// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test chessBoard snapshot with canva 1`] = `"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAboAAAJCCAYAAAC23gJsAAAABmJLR0QA/wD/AP+gvaeTAAAgAElEQVR4nO3de4xU9f3/8dewN13URXflqoiLS79QUCyGX9oUFWUDoV5aK7GtfpW0NfQnKV+TNm0VJbLxa1Fj4iW2Nd/+Gq2aStUCahWKVmuN1J/8Gi7ffttKuYgogiwIKLCzl/P7Yz3DnLMzs3M5M2fO+zwfySZzZmeW92d22efOmZkzCUmOAACw51+S2oaEPQUAAOXmyNa9unYdX5MjKRnuOIFaIu/a9oY7TqCelXdt6R/HJJ0b3mgl2yjvep4Jd5yStEnapezfq9vDGy1wSXnXNivccQLl/75Z0SbvurZIEvfoEAUNkv6voh07C9okvSppTNiDAIUgdKhWPb5tYheubJE7FsIsQEEIHarV3yXt8Z1H7MKRLXK/kvRJ5ccBCkPoUK36JE0QsQtbrsjdWPlxgMIROlSzQyJ2YRoscn0VnwgoAqFDtSN24SByMIPQIQqIXWUROZhC6BAVxK4yiBzMIXSIEmJXXkQOJhE6RA2xKw8iB7MIHaKI2AWLyME0QoeoInbBIHIwj9AhyohdaYgcYoHQIeqIXXGIHGKD0MECYlcYIodYIXSwgtjlh8ghdggdLCF2uRE5xBKhgzXELjMih9gidLCI2HkROcQaoYNVxK4fkUPsETpYFvfYETlAhA72xTV2RA74DKFDHMQtdkQOSEPoEBdxiR2RA3wIHeLEeuyIHJABoUPcWI0dkQOyIHSII2uxI3JADoQOcWUldkQOGAShQ5xFPXZEDsgDoUPcRTV2RA7IE6EDohc7IgcUgNAB/aISOyIHFIjQAcdVe+yIHFAEQgd4VWvsiBxQJEIHDFRtsSNyQAkIHZBZtcSOyAElInRAdmHHjsgBASB0QG5hxY7IAQEhdMDgKh07IgcEiNAB+alU7IgcEDBCB+Sv3LEjckAZEDqgMOWKHZEDyoTQAYULOnZEDigjQgcUJ6jYETmgzAgdULxSY0fkgAogdEBpio0dkQMqhNABpRssdif4zj9JRA6oGEIHBCNX7Np8510oIgdUDKEDgpMtdgnf9om+bSIHlFFt2un20KYI1gW+7SGys7YJvu0G2VnbKN92k6K7tgWSHpU0LI/LrpG0XNKl5RyojBp8259TdL9vfv47Ahdo4B8tVlj5no32bddK/d80p/KzAABQdt2S6tl1CQAwjdABAExLf4yuO7QpgjVEUo3vPCtrq9HAP06srK1WAx//iPLaEur/fg32mE6U1yhJdb7tPkm9YQxSBv619crOE4b8a4v6z6EroSxdc2Trcbp2HV+TIykZ7jiBWiLv2vaGO06gnpV3bRvCHackbZJ2ybuebB/HVP53Ki+nj+Rdz+3hjhOopLxrmxXuOIHy/xxa0SbvurZI7LoEgpbtiCfZlPudyoHYI3RAcLJFbr9vu8u3TeyAMiJ0QDByHbtyl++81Sr/O5UD+AyhA0o32AGa/XpU3ncqB5CG0AGlKfZdCMr1TuUAfAgdULxS32qH2AEVQOiA4gT1fnLEDigzQgcULug3TSV2QBkROqAw5XpncGIHlAmhA/JXrsi5iB1QBoQOyE+5I+cidkDACB0wuEpFzkXsgAAROiC3SkfOReyAgBA6ILuwIucidkAACB2QWdiRcxE7oESEDhioWiLnInZACQgd4FVtkXMRO6BIhA44rloj5yJ2QBEIHdCv2iPnInZAgQgdEJ3IuYgdUABCh7iLWuRcxA7IE6FDnEU1ci5iB+SB0CGuoh45F7EDBkHoEEdWIucidkAOhA5xYy1yLmIHZEHoECdWI+cidkAGhA5xYT1yLmIH+BA6xEFcIucidkAaQgfr4hY5F7EDPkPoYFlcI+cidoAIHeyKe+RcxA6xR+hgEZHzInaINUIHa4hcZsQOsUXoYAmRy43YIZYIHawgcvkhdogdQgcLiFxhiB1ihdAh6ohccYgdYoPQIcqIXGmIHWKB0CGqiFwwiB3MI3SIIiIXLGIH0wgdoobIlQexg1mEDlFC5MqL2MEkQoeoIHKVQexgDqFDFBC5yiJ2MIXQodoRuXAQO5hB6FDNiFy4iB1MIHSoVg0ictVgsNjVVnwioECEDtVqvIhctcgVu2GVHwcoDKFDtarzbRO5cGWLHVD10nc7LAltimBN8G3XyM7aZvm2T5KdtU3J8bm/Stop6bYKzRK00b7tcxXd79v/kfQfkoZm+Xy7pETlximrGt/2fElfCmGOSojqz6Pfab7tE6X+H0in8rMAAFB23ZLq2XUJADCN0AEATEt/jO6j0KYIVoOkU3znWVnbSfpsn/NnHEn7QpolaMPkfQJKn6TOkGYJWrO8f1R2S/o4pFmC1iLvY3LHJB0OaZagne7bPiSpK4xBysC/Niu/I2vkfZzuqHvCka3H6dp1fE2OpGS44wRqibxr2xvuOIF6Vt61bQh3nEBtlHdtz4Q7TqA+kndtt4c7TqCS8q7N/2SwKHN8H1a0ybuuLRK7LgEAxhE6AIBphA4AYBqhAwCYRugAAKYROgCAaYQOAGAaoQMAmEboAACmEToAgGmEDgBgGqEDAJhG6AAAphE6AIBphA4AYBqhAwCYRugAAKYROgCAaYQOAGAaoQMAmEboAACmEToAgGmEDgBgGqEDAJhG6AAAphE6AIBphA4AYBqhAwCYRugAAKYROgCAaYQOAGAaoQMAmEboAACmEToAgGmEDgBgGqEDAJhG6AAAphE6AIBphA4AYBqhAwCYRugAAKYROgCAaYQOAGAaoQMAmEboAACmEToAgGmEDgBgGqEDAJhG6AAAptWmnX42tCmCNdq3XSs7a5vi2x4mO2v7sm/7HNlZ23jf9gzZWVuTb/vfJU0NY5AyqPVtd0j632EMUgFWfh6H+rZPlqSEJKfyswAAUHbdkurZdQkAMI3QAQBMS98HvTG0KYLVJGmc7zwraxsjqSVtu0/S5pBmCdo58u5fT0r6e0izBG2ipPq07U8kbQ1plqBNkfcP5n2S3g9plqCd59veIelgCHOUg39tVn5HNkj6t7Ttw+4JR7Yep2vX8TU56v+FacUSede2N9xxAvWsvGvbEO44gdoo79qeCXecQH0k79puD3ecQCXlXduscMcJlOP7sKJN3nVtkdh1CQAwjtABAEwjdAAA0wgdAMA0QgcAMI3QAQBMI3QAANMIHQDANEIHADCN0AEATCN0AADTCB0AwDRCBwAwjdABAEwjdAAA0wgdAMA0QgcAMI3QAQBMI3QAANMIHQDANEIHADCN0AEATCN0AADTCB0AwDRCBwAwjdABAEwjdAAA0wgdAMA0QgcAMI3QAQBMI3QAANMIHQDANEIHADCN0AEATCN0AADTCB0AwDRCBwAwjdABAEwjdAAA0wgdAMA0QgcAMI3QAQBMI3QAANMIHQDANEIHADCN0AEATCN0AADTCB0AwDRCBwAwLSHJ+ez0pjAHCVCTpLN851lZ22hJLWnbfZL+O6RZgjZe0tC07aSkf4Q0S9D+TVJ92vYnkraFNEvQJsv7B/M+SR+ENEvQpqj/d6Rrh6RD4YwSuHN921Z+R9ar//+ba7+k5vTQAQBgSbekenZdAgBMI3QAANNq3RPLly/fGuYgQTl27NjQgwcPjkw/b8SIESbWdvjw4eYjR44Mc7cTiUTf8OHDt4c5U1D2798/uru7+0R3u6amprulpWVnmDMFZd++fWN7e3vr3O26urqjp512monHsfbu3Xu24zipP5gbGxs/PvnkkzvDnCkoe/bsGZ++3dTU9OEJJ5zwaVjzBMm/Niu/I3fv3u3cfPPN56SdtV9KezKK4/BQHQAgupLJ5LsNDQ3pT0b8l6Q2dl0CAEwjdAAA0wgdAMA0QgcAMI3QAQBMI3QAANMIHQDANEIHADCN0AEATCN0AADTage/CCB1dnbq4MGDYY8RuubmZjU1NVX83+X27xfW7Y9oI3TIS0dHhx588MGwxwjdAw88oEWLFlX83+X27xfW7Y9oY9clAMA07tGhYP7dRzt27FBfX9+Ay40ZM0YNDQ0Dzn/vvffU3d094PyGhgaNGTPGc162XXalzuBKJpPas2ePRowYofr6eh07dkxDhgxRfX39oDOEpdi1O46j7dsHf1en9K+f6/bv6enJebtk282Y6/YcOXKkGhsb8748kC9HkgPksmjRIsf9WXnggQdS599xxx2p8/0fTz/99ICv8+tf/9pJJBIZL19bW+v84Q9/yPrvpn+UMoPjOM6WLVucq666yqmpqXEkOSeeeKJzzTXXOC0tLc5pp53m7NmzZ9C1V1IQt/+qVatS6831kf71c93+2T7nfjz88MODrsX/8eSTT+a9dsCvq6trh+9naovEPTqU4N5779Udd9yR2h49erROOOEESdL555+v9vb2AddZv3596r0Pm5qa1NzcrN7eXu3cuVM9PT36+9//nvF6Qc6wYcMGXXLJJTpw4EDqvKNHj2r58uWp7RUrVmjBggV5zxGGQtf+6quvqre3tyyzuPfe3n//fXV1dUmSTj311LyvJ0lTp07V3LlzyzIf4o3QoSiPP/64fvSjH6W2b7rpJt18882qqamRJJ199tlKJBI5v0ZHR4cWLVqk3bt3D9hlWa4Z+vr6dP3113siN2rUKNXX1+vdd99Nnef+sq5Wxax97ty5Wr58uXbv3i2pPzIHDhzQqaeeOuhuyEwuu+wyrVmzRt3d3VqyZImuvfZatbS0pG67iy66aNCvsXDhQt1www2SpMbGRg0bNqygGYB88GQUFGX9+vWp0/X19XrkkUc0YcIEjR8/XuPHj9fdd9+d99d67bXXinqH+2JmePnll7V582ZJ0rRp03T48GF98MEHeuedd/S1r30tdbl87o2EqZi1t7e3a968eant5uZmjRo1St/97nc1b968VBjr6urymqG9vV3/+Mc/tHXrVt1www1av359KpaTJk3S6NGjB/0aHR0dqZnPOussPffcc3n920AhCB1KlkwmB+wSu+222/SnP/0p5/V2796tP//5z7rnnnsqNsNf/vKX1OnZs2frpJNOktQfi4ceekgzZ87UhRdeqK9//eslz1Qpxd7+77zzjt5//33dfffdSiQSmjx5sqZMmaKLL764qDlefvnl1OlZs2ZlvZx7m/slk0nNmzdv0LmBQhE6FMW/a6+mpkbjxo3TuHHjJEm9vb3auHFjzq+xbNkyXXjhhdqwYUPFZujs7EydHjFihOdzY8aM0SuvvKKHH354wDP/qk0Qt3+6yZMna9OmTdq0aZMmTpxY1Ezpobv00kuzXm7BggWaPXu2WltbUx/u43TJZLKguYF8EDoU5YMPPkidbm5u1ttvv63t27fr97//fc7rXX/99YHtFixmhvTdcpleduDes6l2xd7+6dJvi3x3V2bz6aefat26dZKk2tranPcKx44dq9WrV2vr1q2pD/dxOqAcCB2KsmvXrtTpRx55ROeff/6A8zOZNm2a1q5dq4kTJ6q1tVXnnnvuoE9aCXKGc845J3U6mUwO+PzBgwf1ySefFDVPJRV7+6ebP3++pkyZUtLuStfrr7+euj2nT5+uU045JetlDx8+rJ6enpL+PaAQhA5FSf+FunfvXu3Zs0fbtm1LPb09G8dx9PHHH+uxxx7Txo0bdcYZZ6SeiDJ27Fh99atfLesMc+fOTe2WXLp0qVauXKn33ntPb7zxhm688UYNGzZMo0ePLnp3aqUUs3bHcTzPrBw1apRWrlyplStX5vXEkVzy3W355ptvasSIERo5cqQWL16sdevWadu2bfroo49K+veBXHh5AQrW1dWlffv2pbZvuukm3XTTTXld99FHH9W3v/1tSf27uNy/7EePHq1XXnlFY8eOLesMY8eO1eLFi7V48WJ1dnZ6nmnp+uSTTzLe26sWxa79+eef12OPPZba7ujoUEdHh6TSjyGZ7xNRNm/erKNHj+ro0aO66667dNdddxX9bwL54h4dCnb48GHNnDnTc5gsv9NPPz3jvbOhQ4emTruRO/3007V27VrPbsV8Zrj66qtzvu4q2wy33HJL1jAkEgl1dHRo+vTpec9Sae7ac917y7T2V199Nevljx49WvQ8fX19eueddyT1P+45fPhwbdu2LePr8ubPn6+rr74669caN26crrzyyqJnATIhdChYS0uLJk+enPVez8iRI/XUU08NuHfmOI6mTp2q6667LvUsu4aGBj366KOaNGlSwTMsW7ZMP//5zwcEt66uTq2trRlnkPp3+zU1NWV8HOnOO+/UbbfdVtAsldbS0qLzzjtPx44dy/j5xsbGjGufO3euTjzxxIzXWbJkSdGvYRsyZIjuueceNTU1qaurSxMnTtT48eM99x5dDQ0Nevrpp7VmzRpde+21mjRpkurq6nTGGWdo2rRpeumllzRq1Kii5gCyIXQI3H333adLLrlkwPnPP/+8Jk2apCeeeCL1135XV5e+8Y1v6MiRIwX9G88995wmTJigb37zmwOC29vbqxUrVmScYfv27Tr//PP105/+VIcOHRrw+aVLl1b967gef/zxnDHu6elJHSEl3Ycffpj1nlupr2GbNWtWQfcKW1tbdfToUf3zn/9Ud3e3Ojs7dc4552jGjBkaNWqU9u7dW9QcQCaEDiVrbm5OvR7qqquuynq8wmzHWjx8+LBWrlxZ0L+Z67iNfX19+vGPf5zxc7/97W89r6U77bTT1NramnqCShRex5V+VJRhw4albvvB1lDs9QbjOI4WLlyY9+OaGzZs0PTp0/W73/0u9T10jzW6b98+7d+/XytWrCh4DiAbQoeSLVy4UGvXrtXatWv18MMPZ33cbO7cuZo8ebLOPvvsAZ97/PHHC/o3v/KVr3h2cbmxdXdjrl69OuNjUt/61rc0e/ZsXXHFFXrrrbfU2dmprVu36jvf+U7qMsW+3KFS0o8scv/996deizbYGtIfHy3keoN58skncz7+ly7bsUbPOussz+Wq/VijiBZCh5Lle7zC9vZ2bd68WU888cSAz7388sv68MMP8/43Z82a5Tlu48KFC7Vq1SpdcMEFqfO2bds24HpnnnmmVq9erVWrVmn69Ok6dOiQNm3apNdeey11Gf8v3WqzYMECzZkzR5dffrm++MUvps7funVr6nSmJ9N873vfK+p6uXz88cf64Q9/mNp2j8ySjZVjjSJaCB2KUsrxCtNDV1vb/wqXnp4e/eY3vyl6no6ODk2ZMkVvvvlm6rxHH30059vSvP766xo5cqTOO++81C/f8ePHF/Q2QWEYO3asXnrppdTjlJJ011136cUXX5QkXXnllRmDVez1clm8eLH27Nkjqf8Pmcsvvzzn5S0eaxTVj9ChKMUerzCZTHre9y19t1mme3qleOONN3TnnXfm/Lz/CRTf//73sz4zsVrdc889Wrx4saT+e7pPPfVUXrsgi72ea/369XrkkUck9T/z8gc/+EHGJ/iks3KsUUQLoUNRij1e4Ysvvqj9+/entt1flJL017/+VX/729+Kmqe5uVlnnnmm59mGdXV12rlzp+deRLrrrrtOc+bM8bwe7dZbb43UM/5uueWW1BNvLrroIq1cuVK7du3Sjh07ynK9dLfeemvqHnNfX5/mzJnjeUnBsmXLBnw/rRxrFNFC6FCUYo9XONi9tmLv1S1ZskQ7d+7UgQMH9KUvfUmS1N3drfPOO0+tra0Zr+Puyjtw4EDqWI9HjhzxHOWjmt1yyy1atmyZJGnGjBl64YUXtHTpUs2bNy/na+KKvZ7fYI+j7d69e8AfGVaONYpo4RBgKNi6dev0k5/8RI2NjVqwYIEuu+wyjRgxIufxCjs7O7V792698MILqfOam5vV1NSkvr6+1D2JJ598UjfeeKMkDboLMf3IG52dnaknn6S/EDyRSGj48OGp7V/96leaNm2a7r//fs2YMUNf+MIXdMopp3iewZl+eK1q9bOf/SwVqwkTJuihhx7Sgw8+qHvvvbcs18vk4osvTr1kwXEcbd++3fP5qVOnaubMmZ7z3GONHjlyREuXLtWZZ56padOm6d1339Vjjz2mX/7ylzr55JP1+uuva+rUqQXPBGTjSHKAXBYtWuS4Pyuf//znU6ezfTzwwANZr++/zKpVqzzn19TUOK2trc6mTZsyXk+Sc+mllw46Q319vbN161bPHPfdd5/T1NTkNDY2FjW7/3OVkj7D7NmznZqamoJuf8dxnBdeeKHg62W7/f2XW7VqVcavnWmO//zP/8w5QyKRcN56662MM4R1+yMaurq6dvh+nrZI3KNDES677DI1Njbq7bffzvj5U089VVOmTPE8vT/TcQ/de2H+Fwd/+ctf1tVXX62hQ4dmvJ4k/fGPfxx0TveZe+lzzJw5U7/4xS+0ZcuWkmYP05o1awa9TPo9XNfTTz+d81moma6Xbe3+y61YsSLj1840xzXXXKN169Z57t2nmz9/vlpaWlLXq7bbH9HEPToMKttf9nH7qIZ7dHH+4B4dcsl2j44nowAATGPXJfLiHmIr7tzXClYat3+/sG5/RFtC/XfvUu/yDABAFCWTyXcbGhrOSjvrX5La2HUJADCN0AEATCN0AADTCB0AwDRCBwAwjdABAEwjdAAA0wgdAMC01JFRrrjiitxvDRwRjuPU9fb2et7fpba21sTa+vr6Tujr66tPO8upra09HNpAAert7R3qOE5N2llm1tbT03Oy+g/OIElKJBI9NTU1R0IcKTD+tQ0ZMiQ5ZMiQYyGOFJienp5T0rdramqOJBKJwt+EsQr512bld+Snn37qf5PE4VLakVEAADCmW1I9uy4BAKYROgCAaenvXrAktCmC9TlJ16Zt90m6I5xRAtcuaUba9lFJPw1plqD9u6S2tO19kh4MaZagLZLUkrb9jqQnQpolaLdKOiFt+8+S1oY0S9DukPfOwBPq/95Z0OHbtvL7v1nSf6Rt73VPuG9QZ0W7vG+8lwx3nEAtkXdte3NfPFKelXdtG8IdJ1Ab5V3bM+GOE6iP5F3b7eGOE6ikvGubFe44gfK/qa0VbeKNVwEAcUPoAACmEToAgGmEDgBgGqEDAJhG6AAAphE6AIBphA4AYBqhAwCYRugAAKYROgCAaYQOAGAaoQMAmEboAACmEToAgGmEDgBgGqEDAJhG6AAAphE6AIBphA4AYBqhAwCYRugAAKYROgCAaYQOAGAaoQMAmEboAACmEToAgGmEDgBgGqEDAJhG6AAAphE6AIBphA4AYBqhAwCYRugAAKYROgCAaYQOAGAaoQMAmEboAACmEToAgGmEDgBgGqEDAJhG6AAAphE6AIBphA4AYBqhAwCYRugAAKYROgCAaYQOAGAaoQMAmJaQ5Hx2ujvMQQKUkFTrO8/K2mo08I8TK2urVf/3Lp3VtTmSekKaJWh1vu0+Sb1hDFIG/rX1qn99FvjXZuX/mv/3/xFJQ9NDBwCAJd2S6tl1CQAwjdABAExL35fZHtoUwbpA0k/TtnslzQlplqDNl3Rt2vYhSV8PZ5TAdUj6Ytr2Dkk3hjNK4P5L0ri07XWSloQzSuCelXRK2vYTkh4LaZagrVb/4+Kun0j6fyHNErS1vm0rv/9Hy/vz9757wpGtx+nadXxNjqRkuOMEaom8a9sb7jiBelbetW0Id5xAbZR3bc+EO06gPpJ3bbeHO06gkvKubVa44wTK8X1Y0SbvurZI7LoEABhH6AAAphE6AIBphA4AYBqhAwCYRugAAKYROgCAaYQOAGAaoQMAmEboAACmEToAgGmEDgBgGqEDAJhG6AAAphE6AIBphA4AYBqhAwCYRugAAKYROgCAaYQOAGAaoQMAmEboAACmEToAgGmEDgBgGqEDAJhG6AAAphE6AIBphA4AYBqhAwCYRugAAKYROgCAaYQOAGAaoQMAmEboAACmEToAgGmEDgBgGqEDAJhG6AAAphE6AIBphA4AYBqhAwCYRugAAKYROgCAaYQOAGAaoQMAmEboAACmEToAgGmEDgBgGqEDAJiWkOSEPQQAAGXQLamee3QAANMIHQDANEIHADDPka3H6dp1fE2OpGS44wRqibxr2xvuOIF6Vt61pX/8j6SR4Y1Wso3yrueZcMcpSZukXcr+vbo9vNECl5R3bbPCHSdQ/u+bFW3yrmuLxD06RMNESX9UtGNnQZukVyWNCXsQoBCEDtXqkG+b2IUrW+SOhTALUBBCh2q1Q9ILvvOIXTiyRe5Xkj6p/DhAYQgdqpUj6WoRu7DlityNlR8HKByhQzXrErEL02CR66v4REARCB2qHbELB5GDGYQOUUDsKovIwRRCh6ggdpVB5GAOoUOUELvyInIwidAhaohdeRA5mEXoEEXELlhEDqYROkQVsQsGkYN5hA5RRuxKQ+QQC4QOUUfsikPkEBuEDhYQu8IQOcQKoYMVxC4/RA6xQ+hgCbHLjcghlggdrCF2mRE5xBahg0XEzovIIdYIHawidv2IHGKP0MGyuMeOyAEidLAvrrEjcsBnCB3iIG6xI3JAGkKHuIhL7Igc4EPoECfWY0fkgAwIHeLGauyIHJAFoUMcWYsdkQNyIHSIKyuxI3LAIAgd4izqsSNyQB4IHeIuqrEjckCeCB0QvdgROaAAhA7oF5XYETmgQIQOOK7aY0fkgCIQOsCrWmNH5IAiETpgoGqLHZEDSkDogMyqJXZEDigRoQOyCzt2RA4IAKEDcgsrdkQOCAihAwZX6dgROSBAhA7IT6ViR+SAgBE6IH/ljh2RA8qA0AGFKVfsiBxQJoQOKFzQsSNyQBkROqA4QcWOyAFlRuiA4pUaOyIHVAChA0pTbOyIHFAhhA4o3WCxq/Wdf5KIHFAxhA4IRq7Yjfedd6GIHFAxhA4ITrbYNfi2T/RtEzmgjNJ3qbSHNkWwLvBtD5GdtU3wbTfIztpG+babFN21/VzS6ZL+Vx6XXSNpuaRLyzpR+fgj/jlF9/vm578jcIGkRBiDVICV79lo33at1P9Ncyo/CwAAZdctqZ5dlwAA0wgdAMC09MfoupqsCJEAAARSSURBVEObIlhDJNX4zrOythoN/OPEytpqNfDxjyivLaH+71eux3QcST2VGads6nzbfZJ6wxikDPxr65WdJwz51xbl/2vpEsrSNUe2Hqdr1/E1OZKS4Y4TqCXyrm1vuOME6ll517Yh3HFK0iZpl7zryfbxPyr/O5WX00fyruf2cMcJVFLetc0Kd5xA+X8OrWiTd11bJHZdAkHLdsSTbMr9TuVA7BE6IDjZIrfft73bt03sgDIidEAwch27cpfvvL+o/O9UDuAzhA4o3WAHaPbrU3nfqRxAGkIHlKbYdyEo1zuVA/AhdEDxSn2rHWIHVAChA4oT1PvJETugzAgdULig3zSV2AFlROiAwpTrncGJHVAmhA7IX7ki5yJ2QBkQOiA/5Y6ci9gBASN0wOAqFTkXsQMCROiA3CodORexAwJC6IDswoqci9gBASB0QGZhR85F7IASETpgoGqJnIvYASUgdIBXtUXOReyAIhE64LhqjZyL2AFFIHRAv2qPnIvYAQUidEB0IucidkABCB3iLmqRcxE7IE+EDnEW1ci5iB2QB0KHuIp65FzEDhgEoUMcWYmci9gBORA6xI21yLmIHZAFoUOcWI2ci9gBGRA6xIX1yLmIHeBD6BAHcYmci9gBaQgdrItb5FzEDvgMoYNlcY2ci9gBInSwK+6RcxE7xB6hg0VEzovYIdYIHawhcpkRO8QWoYMlRC43YodYInSwgsjlh9ghdggdLCByhSF2iBVCh6gjcsUhdogNQocoI3KlIXaIBUKHqCJywSB2MI/QIYqIXLCIHUwjdIgaIlcexA5mETpECZErL2IHkwgdooLIVQaxgzmEDlFA5CqL2MEUQodqR+TCQexgBqFDNSNy4SJ2MIHQoVo1iMhVg8Fix+8QVD1+SFGtxovIVYtcsWuq/DhAYQgdqlWdb5vIhStb7GpCmAUoSG3a6bmhTRGsqb7thOysrc23XS87axuR43N/kPQ7SXMqNEvQTvZtj1R0v2+/lDRc0vQsn29TdNfml/BtT1f//zmLrHzPRmc6MyHJqfAgAABUwr8ktbHrEgBgGqEDAJhWK+mlsIcIWIOkKWnbxyT9d0izBK1J3sfpDkraEtIsQRsuaWza9h5J74U0S9DOlPcxyJ2S9oY0S9AmSDolbXuL+n8uLZgs6YS07c3qf1KOBRf4tteHMkX5vR/2AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAALDg/wMKgWSOF4GrFgAAAABJRU5ErkJggg=="`;
